### PR TITLE
feat: select

### PR DESCRIPTION
**Pull request by timcadman**

### CODE REVIEW

**Comment by erikzwart** [comment]
Add short description in README how to get started:
* clone project
* Open dsTidyverse.Rproj in RStudio
* `library(devtools)`
* `devtools::load_all()`

Run tests
* `devtools::test()`

Add basic example for `selectDS()`

**Review by StuartWheater — COMMENTED** [review]
Inspected the code.
Ran devtools::check()
Ran devtools::test()

**Review by StuartWheater — APPROVED** [review]

File: R/utils.R [hunk 1]
```diff
@@ -0,0 +1,138 @@
+#' Generate an encoding key which is used for encoding and decoding strings to pass the R parser
+#'
+#' @return A list containing the encoding key, with 'input' specifying the characters to be encoded
+#' and 'output' specifying their corresponding encoded values.
+#' @noRd
+.getEncodeKey <- function() {
```
Comment (marikaris) [inline R/utils.R:6]: I'm not sure if I like the name of this function. It's not really a key, right? It's more like a dictionary of characters to encode, I think?

File: R/utils.R [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      )
+    }
+  )
+
+  return(object_out)
+}
+
+.permitted_tidy_fun <- function() {
+  permitted <- c("select")
+  return(permitted)
```
Comment (marikaris) [inline R/utils.R:65]: No need to put `c("select")` into a variable, you can return it directly.

File: R/utils.R [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        "x" = "You have attempted to pass {fun}"
+      ),
+      call = NULL
+    )
+  }
+
+  tidy_string <- .make_tidyselect_arg(.data, fun, tidy_select_args)
+  string_as_expr <- rlang::parse_expr(tidy_string)
+  data <- .tidy_eval_handle_errors(string_as_expr, .data)
+  return(data)
```
Comment (marikaris) [inline R/utils.R:92]: Why don't you return the content of data directly without assigning the variable?

File: R/utils.R [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+#' This internal function returns a vector of privacy settings.
+#' @return A character vector containing privacy settings.
+#' @noRd
+.list_privacy_settings <- function() {
+  privacy_settings <- c(
+    "datashield.privacyControlLevel", "nfilter.tab", "nfilter.subset",
+    "nfilter.glm", "nfilter.string", "nfilter.stringShort", "nfilter.kNN",
+    "nfilter.levels.density", "nfilter.levels.max", "nfilter.noise"
+  )
+  return(privacy_settings)
```
Comment (marikaris) [inline R/utils.R:121]: No need to put into a variable, you can return the list directly.

File: R/utils.R [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+#' @param setting The name of the privacy setting.
+#' @return The value of the specified privacy setting.
+#'
+#' @keywords internal
+#'
+.set_privacy_option <- function(setting) {
+  out <- getOption(setting)
+  if (is.null(out)) {
+    out <- getOption(paste0("default.", setting))
+  }
```
Comment (marikaris) [inline R/utils.R:135]: Add test for if case

**Review by marikaris — APPROVED** [review]
Very slight feedback. You may choose to ignore it.

File: R/utils.R (hunk 2 above)
Comment (timcadman) [reply R/utils.R:65]: I've done this because in the future I will add more permitted functions, so will leave as-is

**Pull request merged**